### DISCUSSION TITLE
How to access theme settings from Python plugin?

### DISCUSSION THREAD

**Discussion by ivanipenburg** [Flow-Launcher/Flow.Launcher#1683] [Q&A] [answered]
Hi! I was wondering if there is a way to access the Flow Launcher theme settings from a Python plugin? This would be incredibly useful for making the icons I'm using adapt to the user's theme settings (especially if it's light or dark mode). I'm mainly looking for the text color setting.

**Comment by onesounds** [comment, marked as answer]
Hi, I don't know how, but I don't recommend it. This is because the light/dark state of the flow does not guarantee the color of the theme. (e.g flow's color scheme : dark / flow using theme : win11light)

As I remember, there is no way for the plug-in to adjust the color of the text. Specifying the glyph icon will color it to suit the theme, but I guess you want to something different.

1. I recommend using color icons if possible.
2. If you tell me what you want in detail, I'll see if it's possible.

**Reply by Garulf** [reply]
Flox comes with support to detect the users install location. Iirc there is semi support to grab the launchers settings json that would allow you to read the theme that's set. 

That being said it would probably be best to use glyph icons if possible. There is free software to convert most images to a font you can use.

**Reply by ivanipenburg** [reply]
Thanks for your responses, it sounds like the glyph icon might be what I'm looking for? Could you maybe point me to a place where I could see how they work?

I've also included a mock-up so you can better understand what I want to use it for. Right now those icons for each date are white, but for a light theme user they would turn invisible on a white background, so adapting those icons to the user's theme would be ideal. 

![image](https://user-images.githubusercontent.com/33285651/208410767-2dc2a149-a576-44d7-a9d2-ab14d89cd452.png)

**Reply by Garulf** [reply]
Perfect candidates for Glyphs!

[HA Commander](https://github.com/Garulf/HA-Commander) is a good example on how to utilize custom glyphs. It uses Flox to set a Font Family in the `__init__` method of the plugin.

https://github.com/Garulf/HA-Commander/blob/65a14b3bf6ae03f56b6014656b6bc66ca87ae8fc/plugin/main.py#L31
```
class Commander(Flox, Clipboard):

    def __init__(self):
        super().__init__()
        self.font_family = "#Material Design Icons Desktop"
```
The font_family is the font family name of the font and *not* its filename!


You can set the Glyph by using the `add_item` method like so:

```
self.add_item(
    title=domain,
    subtitle="Search for entities in this domain",
    method=self.change_query,
    parameters=[f"{self.user_keyword} {domain}."],
    dont_hide=True,
    glyph="F0026", 
)
```
You'll need to find out the unicode for your glyph to provide in the glyph argument.

For creating my own font I used [fontforge](https://fontforge.org/en-US/) and provided the file in the plugins root directory.

**Reply by ivanipenburg** [reply]
Sounds exactly like what I was looking for. Thanks for the provided example and the help!

edit:

For future reference, it seems like the glyph code needs to be converted to unicode first. So for the 'Material Design Icon Desktop' example, to get the alert icon with code "F0026", the following code can be used:

```python
self.add_item(
    title=domain,
    subtitle="Search for entities in this domain",
    method=self.change_query,
    parameters=[f"{self.user_keyword} {domain}."],
    dont_hide=True,
    glyph=chr(int("F0026", 16)) 
)
```
